File: aprog.cpp
#include <iostream>
using namespace std;

class Adder {
    int x;
public:
    Adder(int x) : x(x) {}
    int operator()(int y) { return x + y; }
};

Adder outerFunction(int x) {
    return Adder(x);
}

int main() {
    Adder closure = outerFunction(10);
    int result = closure(5); // Returns 15
    cout << result << endl;

    int value = 10;
    // value = "Hello"; // Error
    return 0;
}

File: aprog.py
def outer_function(x):
    def inner_function(y):
        return x + y  # x is captured from the outer scope (closure)
    return inner_function

closure = outer_function(10)
result = closure(5)  # Returns 15
print(result)

# Dynamic typing example
value = 10
print(value)
value = "Hello"  # No error, type changed dynamically
print(value)